### PR TITLE
Support ptex mesh - step 7: Disable texture buffer on Mac OSX

### PR DESCRIPTION
**Pull request by bigbike**
## Motivation and Context
The buffer texture cannot work properly on Mac OSX. We suspect it is due to OpenGL driver shipped with the Mac OSX.
This PR is to add preprocessor directive for conditional compilation on different OS.

-) disabled the bufferTexture in PTexMeshData, Drawable, Shader;
-) disabled the "adjFace" computation;
-) disabled the index computation in shader code;  

## How Has This Been Tested
Test it on Mac:
image

Test it on Linux (to show it does not affect rendering on linux):
image

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bigbike** [comment]
This PR is ready to be reviewed. Thanks.

File: src/esp/assets/PTexMeshData.cpp [hunk 1]
```diff
@@ -797,6 +792,7 @@ void PTexMeshData::uploadBuffersToGPU(bool forceReload) {
     currentMesh->indexBuffer.setData(submeshes_[iMesh].ibo,
                                      Magnum::GL::BufferUsage::StaticDraw);
   }
+#ifndef __APPLE__
```
Comment (erikwijmans) [inline src/esp/assets/PTexMeshData.cpp]:
```suggestion
#ifndef CORRADE_TARGET_APPLE
```

File: src/esp/gfx/PTexMeshDrawable.h [hunk 2]
```diff
@@ -29,7 +29,9 @@ class PTexMeshDrawable : public Drawable {
                     Magnum::SceneGraph::Camera3D& camera) override;
 
   Magnum::GL::Texture2D& atlasTexture_;
+#ifndef __APPLE__
```
Comment (erikwijmans) [inline src/esp/gfx/PTexMeshDrawable.h]:
```suggestion
#ifndef CORRADE_TARGET_APPLE
```

File: src/esp/gfx/PTexMeshShader.cpp [hunk 3]
```diff
@@ -51,6 +53,9 @@ PTexMeshShader::PTexMeshShader() {
 
   vert.addSource(rs.get("ptex-default-gl410.vert"));
   geom.addSource(rs.get("ptex-default-gl410.geom"));
+#ifdef __APPLE__
+  frag.addSource("#define DISABLED_ON_MACOSX\n");
```
Comment (erikwijmans) [inline src/esp/gfx/PTexMeshShader.cpp]:
```suggestion
  frag.addSource("#define CORRADE_TARGET_APPLE\n");
```

**Comment by bigbike** [comment]
Let me know if you have more concerns on this PR.
This is the last major PR to be shipped to support PTex mesh rendering. And we would like to commit it in before the next release date.

**Comment by bigbike** [comment]
Thank all the reviewers.

**Pull request merged**